pepys_import/core/validators/enhanced_validator.py

```diff
@@ -11,40 +11,37 @@
 class EnhancedValidator:
     """Enhanced validator serve to verify the lat/long, in addition to the course/speed/heading"""
 
-    def __init__(self, measurement_object, errors, parser_name):
+    def __init__(self, current_object, errors, parser_name, prev_object=None):
         self.error_type = (
             parser_name + f"-Enhanced Validation Error on Timestamp:"
-            f"{str(measurement_object.time)}, Sensor:"
-            f"{measurement_object.sensor_name}, Platform:{measurement_object.platform_name}"
+            f"{str(current_object.time)}, Sensor:"
+            f"{current_object.sensor_name}, Platform:{current_object.platform_name}"
         )
         self.errors = errors
 
-        if hasattr(measurement_object, "heading"):
-            self.heading = measurement_object.heading
-        else:
-            self.heading = None
-        if hasattr(measurement_object, "course"):
-            self.course = measurement_object.course
-        else:
-            self.course = None
-        if hasattr(measurement_object, "speed"):
-            self.speed = measurement_object.speed
-        else:
-            self.speed = None
-
-        if hasattr(measurement_object, "location"):
-            self.location = measurement_object.location
-        else:
-            self.location = None
-
-        if hasattr(measurement_object, "prev_location"):
-            self.prev_location = measurement_object.prev_location
-        else:
-            self.prev_location = None
-
-        if self.location and self.prev_location:
-            self.course_heading_loose_match_with_location()
-            self.speed_loose_match_with_location()
+        self.heading = (
+            current_object.heading if hasattr(current_object, "heading") else None
+        )
+        self.course = (
+            current_object.course if hasattr(current_object, "course") else None
+        )
+        self.speed = current_object.speed if hasattr(current_object, "speed") else None
+        self.location = (
+            current_object.location if hasattr(current_object, "location") else None
+        )
+        self.time = current_object.time if hasattr(current_object, "time") else None
+
+        if prev_object:
+            self.prev_location = (
+                prev_object.location if hasattr(prev_object, "location") else None
+            )
+            self.prev_time = prev_object.time if hasattr(prev_object, "time") else None
+
+            if self.location and self.prev_location:
+                self.course_heading_loose_match_with_location()
```

Comment:
@BarisSari - does Python have any best practices on passing parameters to methods?  My `traditional` preference would be for the method fields to be passed as parameters, so the inline documentation can detail what the method expects/uses.

But, as I said - are there any pythonic traditions for this?

Reply:
Actually, I'm not sure if there are any pythonic traditions for this. I think we can turn them into static methods and send the location-time variables. In this situation, we don't need instance variables, so I can remove the self keyword from them. Does that make sense?

Reply:
Yes, it would be my preference to make them static - it's a more functional approach, meaning:
* the only data they know about is what was passed in
* the method returns it's results, without any internal changes being made

Reply:
Okay, I'll make them static then.

Reply:
Both Basic and Enhanced Validators are static now (docstrings also added):
https://github.com/debrief/pepys-import/pull/244/commits/cbf0b05c606b0fde7e20cecc182bd1597a7af374
https://github.com/debrief/pepys-import/pull/244/commits/0181848009cf04bb36e19058e24fa6e30b9846e7